plugins/woocommerce-admin/client/products/hooks/use-product-entity-record.ts

```diff
@@ -4,9 +4,32 @@
 import { AUTO_DRAFT_NAME } from '@woocommerce/product-editor';
 import { Product } from '@woocommerce/data';
 import { useDispatch, resolveSelect } from '@wordpress/data';
-
+import { getQuery } from '@woocommerce/navigation';
 import { useEffect, useState } from '@wordpress/element';
 
+type ApiFetchOptions = Record< string, string >;
+
+type Middleware = (
+	options: ApiFetchOptions,
+	next: ( options: ApiFetchOptions ) => void
+) => void;
+
+type ApiFetch = {
+	use: ( middleware: Middleware ) => void;
+};
+
+declare const wp: { apiFetch: ApiFetch };
+
+const isProductEditor = () => {
```

Comment:
For a little extra caution I'm only enabling this only when on the add/edit product screen, since the middle-ware will technically remain as you navigate to other React pages.